cli/cmd/upgrade.go

```diff
@@ -379,6 +385,32 @@ func fetchTLSSecret(k kubernetes.Interface, webhook string, options *upgradeOpti
 	return value, nil
 }
 
+func ensureIssuerCertWorksWithAllProxies(k kubernetes.Interface, cred *tls.Cred) error {
```

Comment:
I think the Linkerd control plane needs to be excluded from this check.  When I run through the cert rotation process, I get this error while replacing the issuer cert:

```
$ ../bin/linkerd upgrade --identity-issuer-certificate-file=./issuer-new.crt --identity-issuer-key-file=./issuer-new.key | kubectl apply -f -
× Failed to build upgrade configuration: You are attempting to use an issuer certificate which does not validate against the trust roots of the following pods:
	* linkerd-controller-8697cfcbd5-fns28
	* linkerd-destination-84979b7dd8-pgdsq
	* linkerd-grafana-7966cbb54-2rrsm
	* linkerd-identity-68c944bbd9-dv2nv
	* linkerd-proxy-injector-777cf54dd8-rw9rw
	* linkerd-sp-validator-8559d89bb7-dp4d8
	* linkerd-tap-5458484f46-nxdwz
	* linkerd-web-56db6684cb-t8dh5
These pods do not have the current trust bundle and must be restarted.  Use the --force flag to proceed anyway (this will likely prevent those pods from sending or receiving traffic).
For troubleshooting help, visit: https://linkerd.io/upgrade/#troubleshooting

error: no objects passed to apply
```

It should not be necessary to manually restart the control plane proxies, since the control plane will automatically get rolled during upgrade anyway.

Reply:
@adleong I think what is happening here is that you have ran these command one after the other in a short period of time and the list of these pods that is displayed here contains the pods that are currently being terminated and replaced with the new pods containing the bundled roots. If you wait a bit longer for them to go away, this will dissapear and you will be allowed to execute the command. In fact if you run the `linkerd check --proxy` cmd at that point you will see that it issues a warning that some proxies do not match the anchors in the config map and if you keep executing the command this list will gradually shrink as part of the proxy restart process. That being said, maybe we should mention it in the docs that users need to wait until the control plane is fully "upgraded" before updating the issuer command. Does that sound correct ? I am not sure we want to hide this error as it gives us extra assurance that the issuer certs work with the proxies of the control plane

Reply:
ah, you're saying that the command to add the new trust root to the bundle should cause the control plane to restart?

Reply:
yes that is correct. You just need to wait a bit until all of that completes. These errors are just caused by the old pods still hanging around. In that case do you think we should change something ? I am not convinced we should